lib/preboot/require.js

```
@@ -19,6 +19,8 @@ module.exports = function (opts) {
       bootable = require(fullpath);
     } catch (ex) { // probably we want to throw an error here and halt the application?
       debug('warning Could not load preboot for %s', fullpath, ex.stack);
+      console.warn('Could not load preboot for %s', fullpath, ex.stack);
+      console.warn('Application startup will continue...');
```

This is duplicating what the `debug` function is doing. Essentially if you add `DEBUG=*` in front of the `node` executable it will display all same information so you would be able to debug the issue locally or even in a certain environment. Just add the ENV var to your container.

---

I think `debug` level is a bit low for something like a preboot failing. I mean really, it should have been a critical error as the comment implies. If an app fails to start one of its components, we should at the very least always see a message telling us what is going on.

---

I agree with Jacob, it probably should have been a critical error. I don't think debug is a matching alternative.
Adding to that, I would argue most devs would not add `DEBUG=*` to understand why their slay app is not behaving as expected and they would be buried in output that will not be relevant if/when they do.

---

Yes I agree the `catch` should throw an error as the the comment on line 20 points out. But the solution here shouldn't be add a `warn` that essentially duplicates what the `debug` message does is my point. The solution IMHO should actually be we throw an error here. Or if we are adding the `console.warn` remove the `debug` method because again it's just duplicative

---

oh yeah I would like that as well!
However, that would be a breaking change, right?

That would probably require a major version bump... and then I think we're in this scenario where it would take most devs a lot of time to be able to upgrade and benefit from this.

I am open to that approach as well though, I just think we could land this first and at the very least provide better diagnostics to most devs while they can set time aside to upgrade.

what do you think?

---

That's fair as long as we can fast follow on this, though Slay needs way more work than just this in a major update 😄